app/src/main/java/com/aconno/sensorics/ui/SensoricsPermissionsHandler.kt

```diff
@@ -0,0 +1,101 @@
+package com.aconno.sensorics.ui
+
+import android.app.AlertDialog
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.provider.Settings
+import android.view.View
+import androidx.core.content.ContextCompat
+import com.aconno.sensorics.BuildConfig
+import com.aconno.sensorics.R
+import com.aconno.sensorics.model.SensoricsPermission
+import com.aconno.sensorics.viewmodel.PermissionViewModel
+import com.google.android.material.snackbar.Snackbar
+import java.lang.IllegalStateException
+
+class SensoricsPermissionsHandler(private val context : Context) : PermissionViewModel.PermissionCallbacks {
+    var contentContainer : View? = null
```

Comment:
This is bad, you are moving View logic out of the View (if you consider Activity as a view).